packages/synapse-react-client/src/components/EntityFinder/tree/EntityTree.tsx

```diff
@@ -206,22 +206,20 @@ export function EntityTree(props: EntityTreeProps) {
     useErrorBoundary: true,
   })
 
-  const {
-    data: projectHeader,
-    isLoading: isLoadingProjectHeader,
-    error: fetchProjectError,
-  } = useGetEntityHeader(projectId!, {
-    enabled: !!(projectId ?? initialContainerPath?.path[1]?.id),
-    refetchInterval: Infinity,
-  })
+  const { data: projectHeader, isLoading: isLoadingProjectHeader } =
+    useGetEntityHeader(projectId!, undefined, {
+      enabled: !!(projectId ?? initialContainerPath?.path[1]?.id),
+      refetchInterval: Infinity,
+    })
 
   if (
     FinderScope.CURRENT_PROJECT === scope &&
     projectId &&
-    fetchProjectError &&
-    fetchProjectError.status === 403
+    !isLoadingProjectHeader &&
+    !projectHeader
   ) {
-    // The user doesn't have access to the project, so let's change the scope to something else
+    // The header wasn't returned, so the user doesn't have access to the current project
+    // Let's change the scope to something else
```

Comment:
`useGetEntityHeader` was changed to allow requesting a version number (which doesn't matter here because this is a project). However, the actual service we use to fetch a versioned entity header doesn't return a 4XX status on unauthorized, because it's a batch API.

I chose not to have two functions that fetch the same information using different services, so we just need to change how we handle the failure case here.

I filed a PLFM ticket to close the existing gap here: https://sagebionetworks.jira.com/browse/PLFM-7989

Reply:
Can't be a 404 since this project ID is determined by looking at the path of an entity, but 401 is definitely possible. Will have to think about this a bit more

Reply:
I don't think we expose the Entity Finder anywhere to unauthenticated users. Even if we do, I think they will likely hit a 401 elsewhere before they hit it here. So while not ideal I think this is acceptable risk